snapshot/android/utils.js

```diff
@@ -12,49 +14,51 @@ const createDirectory = dir => mkdir('-p', dir);
 
 const downloadFile = (url, destinationFilePath) =>
     new Promise((resolve, reject) => {
-        const request = httpsGet(url, response => {
-            switch (response.statusCode) {
-                case 200:
-                    const file = createWriteStream(destinationFilePath, {autoClose: true});
-                    file.on('error', function (error) {
-                        return reject(error);
-                    });
-                    file.on("finish", function() {
+        getRequestOptions(url)
+            .then(options =>
+                get(options)
+                    .on("error", reject)
+                    .pipe(createWriteStream(destinationFilePath, { autoClose: true }))
+                    .on("finish", _ => {
                         chmodSync(destinationFilePath, 0755);
                         return resolve(destinationFilePath);
-                    });
-                    response.pipe(file);
-                    break;
-                case 301:
-                case 302:
-                case 303:
-                    const redirectUrl = response.headers.location;
-                    return this.downloadExecFile(redirectUrl, destinationFilePath);
-                default:
-                    return reject(new Error("Unable to download file at " + url + ". Status code: " + response.statusCode));
-            }
-        });
+                    })
+            ).catch(reject);
+    });
 
-        request.end();
+const getJsonFile = url =>
+    new Promise((resolve, reject) => {
+        getRequestOptions(url)
+            .then(options =>
+                get(options, (error, response, body) => {
+                    if (error) {
+                        return reject(error);
+                    }
+
+                    if (!response || response.statusCode !== 200) {
+                        return reject(`Couldn't fetch ${url}! Response:\n${response}`);
+                    }
 
-        request.on('error', function(err) {
-            return reject(err);
-        });
+                    try {
+                        const data = JSON.parse(body);
+                        resolve(data);
+                    } catch (error) {
+                        reject(`Couldn't parse json data! Original error:\n${error}`);
+                    }
+                })
+            ).catch(reject);
     });
 
-const getJsonFile = url =>
+const getRequestOptions = (url) =>
     new Promise((resolve, reject) => {
-        httpsGet(url, res => {
-            let body = "";
-            res.on("data", chunk => {
-                body += chunk;
+        const options = { url };
```

Comment:
what about followAllRedirects option? Do you think we should add it by default?

Reply:
See:
https://github.com/NativeScript/nativescript-dev-webpack/pull/428#discussion_r167865154